src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -276,3 +340,27 @@
                                          :account-name  account-name}))}]]]}
          (not transaction-confirmed?)
          (assoc :db (update-in db [:wallet :ui :swap] dissoc :approval-transaction-id)))))))
+
+(rf/reg-event-fx :wallet.swap/swap-transaction-update
+ (fn [{:keys [db]} [{:keys [tx-hash status]}]]
+   (let [{:keys [pay-amount pay-token-symbol
+                 receive-amount receive-token-symbol]} (get-in db
+                                                               [:wallet :transactions tx-hash
+                                                                :swap-data])
+         transaction-confirmed-or-failed?              (#{:confirmed :failed} status)
```

Comment:
What is this sorcery 😅

Reply:
Just checking if status of the transaction is the final one (either confirmed or failed). Say this pattern in many places of the codebase, for example this: https://github.com/status-im/status-mobile/blob/35e71289dce758ba04e3772cf64bbf63bef59dd2/src/status_im/contexts/profile/settings/events.cljs#L28